core/new-gui/src/app/workspace/service/workflow-graph/model/workflow-graph.ts

```diff
@@ -707,6 +758,22 @@ export class WorkflowGraph {
     updateYTypeFromObject(previousProperty, newProperty);
   }
 
+  public setPortProperty(operatorPortID: OperatorPort, newProperty: object) {
+    newProperty = newProperty as PortProperty;
+    if (!this.hasPort(operatorPortID))
+      throw new Error(`operator port ${(operatorPortID.operatorID, operatorPortID.portID)} does not exist`);
+    const portDescriptionSharedType = this.getSharedPortDescriptionType(operatorPortID);
+    if (portDescriptionSharedType === undefined) return;
+    portDescriptionSharedType.set(
+      "partitionRequirement",
+      createYTypeFromObject<PartitionInfo>((newProperty as PortProperty).partitionInfo) as unknown as PartitionInfo
```

Comment:
I am new to this syntax `as unknown as PartitionInfo`, could you elaborate what is the purpose of it?